### ISSUE TITLE
Primary outlet navigation canceled while navigating in another outlet

### ISSUE THREAD

**Issue by marsc** [angular/angular#27502]
# 🐞 bug report

### Affected Package
"@angular/router": "^7.1.1"

### Is this a regression?
No.

### Description
Navigating to a route in the primary outlet while navigating to another route in a named outlet cancels the primary outlet navigation.

Navigating in one outlet should not affect other outlets.

## 🔬 Minimal Reproduction
https://stackblitz.com/edit/angular-amx5in

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 7.1.1
Node: 10.14.1
OS: darwin x64
Angular: 7.1.1
... animations, cdk, cli, common, compiler, compiler-cli, core
... forms, http, language-service, material, platform-browser
... platform-browser-dynamic, router, upgrade

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.11.1
@angular-devkit/build-angular     0.11.1
@angular-devkit/build-optimizer   0.11.1
@angular-devkit/build-webpack     0.11.1
@angular-devkit/core              7.1.1
@angular-devkit/schematics        7.1.1
@ngtools/webpack                  7.1.1
@schematics/angular               7.1.1
@schematics/update                0.11.1
rxjs                              6.3.3
typescript                        3.1.6
webpack                           4.23.1
</code></pre>

**Comment by jasonaden** [comment]
This issue doesn't give any details of how you set up this test or what you want me to do. Also, what are you expecting to happen and what's happening that doesn't make sense? Please update to include these details.

**Comment by marsc** [comment]
@jasonaden Sorry if this is unclear. 

Always start from "Home". If you click on "Single Routing" the navigation to SuccessComponent is of course successful and you will see "Routing successful". But if you click on "Parallel Routing" The navigation to SuccessComponent gets canceled. (See console)

`NavigationCancel {id: 4, url: "/success", reason: "Navigation ID 4 is not equal to the current navigation id 5"}`

This happens only if the **RouteGuards.canActivate** is still running when the 2nd navigation starts. If there is no guard or the guard is already done, the navigation won't be canceled.

I expect that the navigation won't be canceled if I navigate in another outlet only.

`// SuccessComponent, CanActivate
this.router.navigate( [ 'success' ] );
// 2nd navigation, which causes the 1st navigation to be canceled
this.router.navigate( [ { outlets: { modal: null } } ] );`

**Comment by jasonaden** [comment]
Okay, thanks for the clarification.

As designed, this is expected behavior. The reasoning here is we need to confirm activation of any running navigations before we can ensure the new navigation will be successful. In some applications, the "parallel route" might be completely separate from the parents. But in others, or when the parallel route is deeply nested, if the previous navigation were to fail you might be targeting somewhere that could never be reached.

I think the reason this works after `canActivate` finishes is that everything after that can run synchronously. Therefore you will get a successful navigation, including activation, all the way to the end.

However, if you were to add a resolver that takes some time, I would be willing to bet you could kick off a new navigation while that's running and you would see the same cancellation behavior.

Does that make sense? I'm leaving this open so we can finish off the discussion here, but plan to close this as "Works as intended".

**Comment by jasonaden** [comment]
Marking this "Works as intended" as mentioned above. Please create a new issue if there's more to be looked into here.
